test/sol5/kyberNetworkProxy.js

```diff
@@ -578,6 +578,36 @@ contract('KyberNetworkProxy', function(accounts) {
             });
         }
     });
+
+    describe("test swap function (backward compatible)", async () => {
+        it("swapEtherToToken", async () => {
+            let initialTakerBalance = await tokens[1].balanceOf(taker);
+            let tokenAdd = tokens[1].address;
+            let srcQty = (new BN(2)).mul((new BN(10)).pow(ethDecimals));
+            let proxyRate = await networkProxy.getExpectedRate(ethAddress, tokenAdd, srcQty);
+            await networkProxy.swapEtherToToken(tokenAdd, proxyRate.worstRate, { from: taker, value: srcQty });
```

Comment:
in general worstRate usage is not good. its only stayed as backward compatible value.
in new getExpectedRate we don't return this value.
So most of the tests should use actual returned rate (not worst rate).

it is important to test this in 1-2 tests.
make sure its always lower then expected rate.
please do verify its always rate *  97 / 100

Reply:
change as request